### PR TITLE
Add and update routes needed for linking home page

### PR DESCRIPTION
**Pull request by jmac522**
Changes included: 
- Getting all users 
- Getting all albums 
- Getting reviews written by the followings of a specific user 
- updated isModerator to isMod per julia's request

### CODE REVIEW

File: db/controllers/album-controller.js [hunk 1]
```diff
@@ -146,4 +148,35 @@ albumRouter.get("/search/:query", async (req, res) => {
...
+                duplicating: false,
+            },
+        ],
+        attributes: [
+            "id",
+            "spotifyId",
+            "image",
+            "name",
+            "artist",
+            [sequelize.fn("COUNT", sequelize.col("Likes.AlbumId")), "likesCount"],
```
Comment (jmac522) [inline db/controllers/album-controller.js:168]: This includes the number of likes the album has with the response

File: db/controllers/like-controller.js [hunk 2]
```diff
@@ -76,7 +77,16 @@ likeRouter.get('/liked-albums/:userId', async (req, res) => {
...
                 }
-            ]
+            ],
+            attributes: [
+                "id",
+                "spotifyId",
+                "image",
+                "name",
+                "artist",
+                [sequelize.fn("COUNT", sequelize.col("Likes.AlbumId")), "likesCount"],
```
Comment (jmac522) [inline db/controllers/like-controller.js:87]: Including the likesCount for users' liked albums

File: db/controllers/review-controller.js [hunk 3]
```diff
@@ -8,36 +8,36 @@ const reviewRouter = express.Router();
...
+        res.status(500).send({
+            message: "Some error occurred while creating or updating the Review.",
+        });
     }
 });
 
 reviewRouter.get("/", (req, res) => {
     // Get all reviews
-    Review.findAll()
+    Review.findAll({include: [User, Album]})
```
Comment (jmac522) [inline db/controllers/review-controller.js:40]: Needed to include the User and Album object to have access to the necessary info to display the review on the home page

File: db/controllers/album-controller.js (hunk 1 above)
Comment (hunterrudnet) [reply db/controllers/album-controller.js:168]: very nice

Comment (hunterrudnet) [reply db/controllers/album-controller.js:168]: looks good!

File: db/controllers/like-controller.js (hunk 2 above)
Comment (hunterrudnet) [reply db/controllers/like-controller.js:87]: cool!

File: db/controllers/review-controller.js (hunk 3 above)
Comment (hunterrudnet) [reply db/controllers/review-controller.js:40]: This makes sense

**Review by hunterrudnet — APPROVED** [review]
LGTM. Please add the example calls to the postman if you can!

**Pull request merged**